samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -93,6 +93,11 @@ public StreamOperatorTask(OperatorSpecGraph specGraph) {
   public final void init(Context context) throws Exception {
     // create the operator impl DAG corresponding to the logical operator spec DAG
     this.operatorImplGraph = new OperatorImplGraph(specGraph, context, clock);
+    /*
+     * TODO: Consolidate the thread pool used by OperatorImpl and StreamOperatorTask. For now, we need to keep the
+     * existing behavior. The refactor however is to clean up the DI which is one step towards consolidation
+     */
+    this.taskThreadPool = context.getContainerContext().getContainerThreadPool();
```

Comment:
why is the task executor using the container thread pool?

Reply:
It always used container thread pool prior to this refactor as well. There was only one framework thread pool for historic context and hence it always used that one.